tests/run_tests.py

```diff
@@ -189,11 +208,14 @@ def add_test_methods(cls):
     testset = [testptn.match(e).group('name') for e in listdir(TESTS) if testptn.match(e)]
     testset = sorted(set(testset))
 
-    for filename in testset:
+    for idx, filename in enumerate(testset):
       methname = 'test_' + filename
       def create_closure():
         name = filename
-        return lambda self: self._run_test(name)
```

Comment:
Fixes a deprecation warning in 3.12: `unittest/case.py:690: DeprecationWarning: It is deprecated to return a value that is not None from a test case (<bound method add_test_methods.<locals>.create_closure.<locals>.<lambda> of <__main__.TestPatchFiles testMethod=test_11permission>>)`